pkg/cluster/audit.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"os"
+
+	"github.com/coreos/etcd-operator/pkg/spec"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+func NewAuditLogger(cl *spec.Cluster, lg *logrus.Entry) *logrus.Logger {
+	if cl.Spec.SelfHosted == nil {
+		return nil
+	}
+
+	mountPath := "/var/tmp/etcd-operator/"
```

Comment:
@xiang90 Actually the `OperatorRoot` filepath is not mounted/created before the debug logger is setup.
The `OperatorRoot` path is created after that by the backup manager during the [cluster setup](https://github.com/coreos/etcd-operator/blob/master/pkg/cluster/cluster.go#L109) by the backup manager while creating the credentials and config files from the aws secret.
https://github.com/coreos/etcd-operator/blob/master/pkg/cluster/backupstorage/s3.go#L35-L39 

Although I agree an explicit flag to enable debugging is better than silently detecting the presence of the mountPath.

My only concern now is that when you mount a hostPath to `/var/tmp/etcd-operator/` for debugging, and if you set up S3 backup, the credentials and config files will be written to the hostPath as well now. Even though they are [deleted later](https://github.com/coreos/etcd-operator/blob/master/pkg/cluster/backupstorage/s3.go#L91) when the cluster is destroyed successfully, just wondering if this is alright.